src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs

```diff
@@ -66,10 +66,13 @@ internal partial class AbstractSyntaxIndex<TIndex> : IObjectWritable
 
                 // attempt to load from persisted state
                 using var stream = await storage.ReadStreamAsync(documentKey, s_persistenceName, checksum, cancellationToken).ConfigureAwait(false);
-                using var gzipStream = new GZipStream(stream, CompressionMode.Decompress, leaveOpen: true);
-                using var reader = ObjectReader.TryGetReader(stream, cancellationToken: cancellationToken);
-                if (reader != null)
-                    return read(stringTable, reader, checksum);
+                if (stream != null)
```

Comment:
stream can be null when no data is stored.  didn't test the 'load first then save' scenario.